### PR TITLE
Branch 3.0.x

### PR DESCRIPTION
**Pull request by guljain**

### CODE REVIEW

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GhfsStorageStatistics.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
 public class GhfsStorageStatistics extends StorageStatisticsFromIOStatistics {
 
   /** {@value} The key that stores all the registered metrics */
-  public static final String NAME = "GhfsStorageStatistics";
+  public static final String NAME = "GhfsFileSystemBasedStorageStatistics";
```
Comment (arunkumarchacko) [inline gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GhfsStorageStatistics.java:29]: If we change the name, will the plugin get hold of the statistics?

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GhfsStreamStats.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.hadoop.fs.gcs;
+
+import javax.annotation.Nonnull;
+
+class GhfsStreamStats {
```
Comment (arunkumarchacko) [inline gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GhfsStreamStats.java:21]: I know we did not do this in 2.2. But a comment explaining what this class is doing will be useful.

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystem.java [hunk 3]
```diff
@@ -218,18 +221,40 @@ public class GoogleHadoopFileSystem extends FileSystem implements IOStatisticsSo
...
+    GoogleCloudStorageEventBus.register(globalStorageStatistics);
+  }
 
   /**
    * Constructs an instance of GoogleHadoopFileSystem using the provided
    * GoogleCloudStorageFileSystem; initialize() will not re-initialize it.
    */
   @VisibleForTesting
   GoogleHadoopFileSystem(GoogleCloudStorageFileSystem gcsfs) {
+    this();
```
Comment (arunkumarchacko) [inline gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystem.java:257]: is this required?

**Review by arunkumarchacko — APPROVED** [review]

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GhfsStorageStatistics.java (hunk 1 above)
Comment (guljain) [reply gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GhfsStorageStatistics.java]: Plugin will get Name for GhfsGlobalStorageStatistics and not this class.

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GhfsStreamStats.java (hunk 2 above)
Comment (guljain) [reply gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GhfsStreamStats.java]: Done

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystem.java (hunk 3 above)
Comment (guljain) [reply gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystem.java]: Yes this is required to setup global storage statistics

**Pull request closed without merging**